packages/core/test/utils.spec.ts

```diff
@@ -33,9 +34,55 @@ describe('utils', () => {
     })
 
     it('should merge arrays', () => {
-      const target = [{ a: 'target' }, 'extraString'] as any
-      const source = [{ b: 'source' }] as any
-      expect(combine(target, source)).toEqual([{ a: 'target', b: 'source' }, 'extraString'])
+      const sourceA = [{ a: 'source a' }, 'extraString'] as any
+      const sourceB = [{ b: 'source b' }] as any
+      expect(combine(sourceA, sourceB)).toEqual([{ a: 'source a', b: 'source b' }, 'extraString'])
+    })
+
+    it('should merge multiple objects', () => {
+      expect(combine({ a: 1 }, { b: 2 }, { c: 3 })).toEqual({ a: 1, b: 2, c: 3 })
+    })
+
+    it('should not keep references on objects', () => {
+      const source = { a: { b: 1 } }
+      const result = combine({}, source)
+      source.a.b = 2
+      expect(result.a.b).toBe(1)
+    })
+
+    it('should not keep references on arrays', () => {
+      const source = { a: [1] }
+      const result = combine({}, source)
+      source.a[0] = 2
+      expect(result.a[0]).toBe(1)
+    })
+  })
+
+  describe('deepClone', () => {
+    it('should return a similar value', () => {
+      const clonedValue = deepClone({ a: 1 })
+      expect(clonedValue).toEqual({ a: 1 })
+    })
+
+    it('should clone the value', () => {
+      const value = { a: 1 }
+      const clonedValue = deepClone(value)
+      value.a = 2
+      expect(clonedValue).toEqual({ a: 1 })
+    })
```

Comment:
what about `should return a different reference` instead?
I think it could be clearer to be explicit about what clone mean in our tests description